evalml/tests/automl_tests/test_automl.py

```diff
@@ -623,6 +624,52 @@ def test_data_splitter_shuffle():
 
 
+@pytest.mark.parametrize("automl_type", [ProblemTypes.BINARY, ProblemTypes.MULTICLASS, ProblemTypes.REGRESSION])
+@patch('evalml.pipelines.RegressionPipeline.score')
+@patch('evalml.pipelines.RegressionPipeline.fit')
+@patch('evalml.pipelines.MulticlassClassificationPipeline.score')
+@patch('evalml.pipelines.MulticlassClassificationPipeline.fit')
+@patch('evalml.pipelines.BinaryClassificationPipeline.score')
+@patch('evalml.pipelines.BinaryClassificationPipeline.fit')
+def test_data_splitter_gives_pipelines_same_data(mock_fit_binary, mock_score_binary,
+                                                 mock_fit_multi, mock_score_multi,
+                                                 mock_fit_regression, mock_score_regression,
+                                                 automl_type, caplog,
+                                                 X_y_binary, X_y_multi, X_y_regression):
+    if automl_type == ProblemTypes.BINARY:
+        X, y = X_y_binary
+        mock_score_binary.return_value = {'Log Loss Binary': 1.0}
+        mock_fit = mock_fit_binary
+        mock_score = mock_score_binary
+    elif automl_type == ProblemTypes.MULTICLASS:
+        X, y = X_y_multi
+        mock_score_multi.return_value = {'Log Loss Multiclass': 1.0}
+        mock_fit = mock_fit_multi
+        mock_score = mock_score_multi
+    elif automl_type == ProblemTypes.REGRESSION:
+        X, y = X_y_regression
+        mock_score_regression.return_value = {'R2': 1.0}
+        mock_fit = mock_fit_regression
+        mock_score = mock_score_regression
+
+    automl = AutoMLSearch(X_train=X, y_train=y, problem_type=automl_type, max_batches=1, n_jobs=1)
+    automl.search()
+    n_pipelines_evaluated = len(automl.results['pipeline_results'])
+    assert n_pipelines_evaluated > 1
+    # current automl algo trains each pipeline using 3-fold CV for "small" datasets (i.e. test data above)
+    # therefore, each pipeline should recieve an identical set of three training-validation splits
+    # we'll check the first couple to validate that the data splitter is being used correctly
+    for fold_num in range(3):
+        pipeline0_training_X, pipeline0_training_y = mock_fit.call_args_list[fold_num].args
+        pipeline1_training_X, pipeline1_training_y = mock_fit.call_args_list[3 + fold_num].args
+        pipeline0_validation_X, pipeline0_validation_y = mock_score.call_args_list[fold_num].args
+        pipeline1_validation_X, pipeline1_validation_y = mock_score.call_args_list[3 + fold_num].args
+        assert joblib_hash(pipeline0_training_X.to_dataframe()) == joblib_hash(pipeline1_training_X.to_dataframe())
+        assert joblib_hash(pipeline0_training_y.to_series()) == joblib_hash(pipeline1_training_y.to_series())
+        assert joblib_hash(pipeline0_validation_X.to_dataframe()) == joblib_hash(pipeline1_validation_X.to_dataframe())
+        assert joblib_hash(pipeline0_validation_y.to_series()) == joblib_hash(pipeline1_validation_y.to_series())
```

Comment:
This is a simple version of the great reproducer @freddyaboulton included in #1982 !